media/js/newsletter/newsletter.es6.js

```diff
@@ -1,214 +0,0 @@
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at https://mozilla.org/MPL/2.0/.
- */
-
-import FormUtils from './form-utils.es6';
-
-let form;
-
-const NewsletterForm = {
-    handleFormError: (msg) => {
-        let error;
-
-        FormUtils.enableFormFields(form);
-
-        form.querySelector('.mzp-c-form-errors').classList.remove('hidden');
-
-        switch (msg) {
-            case FormUtils.errorList.EMAIL_INVALID_ERROR:
-                error = form.querySelector('.error-email-invalid');
-                break;
-            case FormUtils.errorList.NEWSLETTER_ERROR:
-                form.querySelector(
-                    '.error-newsletter-checkbox'
-                ).classList.remove('hidden');
-                break;
-            case FormUtils.errorList.COUNTRY_ERROR:
-                error = form.querySelector('.error-select-country');
-                break;
-            case FormUtils.errorList.LANGUAGE_ERROR:
-                error = form.querySelector('.error-select-language');
-                break;
-            case FormUtils.errorList.PRIVACY_POLICY_ERROR:
-                error = form.querySelector('.error-privacy-policy');
-                break;
-            case FormUtils.errorList.LEGAL_TERMS_ERROR:
-                error = form.querySelector('.error-terms');
-                break;
-            default:
-                error = form.querySelector('.error-try-again-later');
-        }
-
-        if (error) {
-            error.classList.remove('hidden');
-        }
-    },
-
-    handleFormSuccess: () => {
-        const newsletters = Array.from(
-            document.querySelectorAll("input[name='newsletters']:checked")
-        ).map((newsletter) => newsletter.value);
-
-        form.classList.add('hidden');
-        document.getElementById('newsletter-thanks').classList.remove('hidden');
-
-        if (window.dataLayer) {
-            // UA
-            window.dataLayer.push({
-                event: 'newsletter-signup-success',
-                newsletter: newsletters
-            });
-            // GA4
-            for (let i = 0; i < newsletters.length; ++i) {
-                window.dataLayer.push({
-                    event: 'newsletter_subscribe',
-                    newsletter_id: newsletters[i]
-                });
-            }
-        }
-
-        // Glean
-        if (typeof window.Mozilla.Glean !== 'undefined') {
-            window.Mozilla.Glean.pageEvent({
-                label: 'newsletter-sign-up-success',
-                type: newsletters.join(', ')
-            });
-        }
```

Comment:
We need to preserve these analytics events when a newsletter is successfully submitted. 

I added a callback you can implement here to help with this: https://protocol.mozilla.org/components/detail/newsletter--default

Reply:
Hmm, one thing I didn't do is pass an object containing which newsletters we're subscribed to (we should probably do that in a future update). But for now you should be able to set the newsletter IDs by querying the form directly.

Let me know if you need a hand here!